### ISSUE TITLE
CORS Error

### ISSUE THREAD

**Issue by leelandmiller** [serverless-components/fullstack-app#14]
Just installed the fullstack-app and followed the instructions to add the API url to my frontend's config.
But I am getting a CORS error out of the box.

`Access to fetch at '{apiUrl}/users/register' from origin '{cloudfrontUrl}' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

I have not touched any of the boilerplate code yet at all. Just added the API url, and deployed

**Comment by austencollins** [comment]
Thanks for reporting.  Looking into this now.

**Comment by leelandmiller** [comment]
Thanks for the fast response!
I don't know if this helps, but I poked around in API Gateway on my AWS console and found this

<img width="1505" alt="Screen Shot 2020-08-10 at 4 40 14 PM" src="https://user-images.githubusercontent.com/7775399/89841620-4d489a80-db28-11ea-9454-303df4cf97ea.png">

Does CORS need to be set in `serverless.yml` for the express api as well?

**Comment by leelandmiller** [comment]
Just wanted to follow up. I was able to fix this by manually changing the CORS settings from the AWS Console:
<img width="1488" alt="Screen Shot 2020-08-12 at 4 56 23 PM" src="https://user-images.githubusercontent.com/7775399/90079468-de507a80-dcbc-11ea-92df-2d766f7f8e20.png">
Is there anything that can be added to `serverless.yml` to set these CORS configurations at deploy when using the new Express Component?

Also in `api/app.js` I did change the CORS middleware from `res.header` to `res.set` per express.js' docs
<img width="358" alt="Screen Shot 2020-08-12 at 4 57 40 PM" src="https://user-images.githubusercontent.com/7775399/90079579-24a5d980-dcbd-11ea-816f-f9c967ebec5c.png">

**Comment by austencollins** [comment]
Hi @leelandmiller,

Sorry for the slow follow-up.  This is caused by an uncaught error, which I have pushed some fixes for, and created a guide [here](https://github.com/serverless-components/express/blob/master/README.md#how-to-debug-cors-errors) to understand this issue better.

TLDR, the code was generating an error that was not caught correctly, and therefore the API response did not include the CORS headers.  You **do not** have to configure your HTTP API on AWS to handle CORS.  You can simply return the standard CORS headers from your code to enable CORS.  No IaaS config changes needed.  However, you must capture all errors correctly and return responses with the correct headers.

**Comment by leelandmiller** [comment]
No apology necessary. Not slow by any means!!
Thanks @ac360 for all that you and your team have put into the Serverless framework.

**Comment by austencollins** [comment]
Thanks for the great feedback @leelandmiller 👍
